Source/SwiftLintFramework/Rules/ExplictSelfRule.swift

```diff
@@ -0,0 +1,286 @@
+    )
+
+    public func validateFile(
+        file: File,
+        kind: SwiftDeclarationKind,
+        dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+
+        //only check kinds that might require `self.`
+        let types: [SwiftDeclarationKind] = [
+            .FunctionMethodInstance, .VarInstance,
+            .FunctionMethodStatic, .VarStatic,
+        ]
+        guard types.contains(kind) else { return [] }
+
+        //build a set of instance/static memebers
+        let memberTypes: [SwiftDeclarationKind] = [
+            .FunctionMethodInstance, .VarInstance,
+            .FunctionMethodStatic, .VarStatic,
+        ]
+        let instanceMembers = file.members(memberTypes)
+        let instanceMemberNames = instanceMembers.flatMap { $0["key.name"] as? String }
+
+        //get the members body (functions and calculated properties will have this data)
+        guard
+            let bodyLocation = (dictionary["key.bodyoffset"] as? Int64).flatMap({ Int($0) }),
+            let bodyLength = (dictionary["key.bodylength"] as? Int64).flatMap({ Int($0) })
+            else { return [] }
+
+        //get all the tokens inside the body
+        let range = NSRange(location: bodyLocation, length: bodyLength)
+        let tokens = file.syntaxMap.tokensIn(range)
+
+        //get parameters (if any)
+        let parameters = file
+            .members(dictionary, declarations: [SwiftDeclarationKind.VarParameter])
+            .flatMap { $0["key.name"] as? String }
+
+        var violations: [StyleViolation] = []
+
+        var previous: [String] = []
+        for token in tokens {
+            guard let type = SyntaxKind(rawValue: token.type) else { continue }
+
+            let value = token.name(file)
+            defer { previous.append(value) }
+
+            let allowedSuffixes: [[String]] = [
+                ["self"], //self.value (normal access)
+                ["let"], //'let value' or 'if let value' - local scope/unwrap shadowing
+                ["_"], //_ = value - local scope shadowing
+            ]
+
+            let isInstanceIdentifier = (type == .Identifier &&
+                token.matches(instanceMemberNames, file: file))
+
+            let isShadowingParameter = parameters.contains(value)
+
+            let isViolation = (isInstanceIdentifier && !isShadowingParameter
+                && !previous.suffix(matches: allowedSuffixes))
+
+            guard isViolation else { continue }
+
+            violations.append(StyleViolation(
+                ruleDescription: self.dynamicType.description,
+                severity: configuration.severity,
+                location: Location(file: file, byteOffset: token.offset)
+                )
+            )
+        }
+        return violations
+    }
+}
+
+extension SyntaxToken {
+    func name(file: File) -> String {
+        return file.contents.substring(self.offset, length: self.length)
```

Comment:
Even though that's the opposite of what this rule is enforcing, this project prefers omitting `self.` wherever possible. Could you please adapt your code to be consistent with this?